lib/Infer/AliveDriver.cpp

```diff
@@ -946,23 +957,33 @@ std::unordered_set<RefinementProblem, RefinementProblem::Hash>
   std::unordered_set<RefinementProblem, RefinementProblem::Hash> Result;
 
   for (auto Change : ChangeList) {
-    Result.insert(P.ReplacePhi(IC, Change));
+    auto Goal = P.ReplacePhi(IC, Change);
+    // Consider switching to better data structures for dealing with BPCs
+    for (auto [Block, Pred] : Change) {
+      for (BlockPCMapping BPC : Goal.BPCs) {
+        if (BPC.B == Block && BPC.PredIdx == Pred) {
+          auto Ante = IC.getInst(Inst::Eq, 1, {BPC.PC.LHS, BPC.PC.RHS});
+          Goal.Pre = IC.getInst(Inst::And, 1, {Goal.Pre, Ante});
+        }
+      }
+    }
+    Result.insert(Goal);
   }
-
   return Result;
 }
 
 }
 bool souper::isTransformationValid(souper::Inst* LHS, souper::Inst* RHS,
                                    const std::vector<InstMapping> &PCs,
+                                   const souper::BlockPCs & BPCs,
```

Comment:
eliminate space between & and BPcs

Reply:
also, a couple lines above, always put the "*" next to the variable name, not next to the type, since in some circumstances that is misleading